src/tests/unit/infrastructure/collection/clients/test_semantic_scholar_client.py

```
@@ -0,0 +1,143 @@
+    mock_response_429 = Mock()
+    mock_response_429.status_code = 429
+
+    exc = requests.exceptions.HTTPError("429 Too Many Requests")
+    exc.response = mock_response_429
+    mock_http.get.side_effect = exc
+
+    client = SemanticScholarClient(api_key=None, http_client=mock_http)
+
+    try:
+        client.fetch_papers(query="test")
+        assert False, "Expected SemanticScholarRateLimitedError"
```

Ruff (0.15.15)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)